app/approot.go

```diff
@@ -147,21 +150,153 @@ func getCumulativeCapitalGains(deltasBySec map[string]*SecurityDeltas) *AllCumul
 		securityGains[sec] = ptf.CalcSecurityCumulativeCapitalGains(deltas.Deltas)
 	}
 	aggregateGains := ptf.CalcCumulativeCapitalGains(securityGains)
-	return &AllCumulativeCapitalGains{securityGains, aggregateGains}
+	return &AllCumulativeCapitalGains{
+		SecurityGains:  securityGains,
+		AggregateGains: aggregateGains,
+	}
 }
 
 type AppRenderResult struct {
 	SecurityTables      map[string]*ptf.RenderTable
 	AggregateGainsTable *ptf.RenderTable
 }
 
+func mustDecimal(opt decimal_opt.DecimalOpt) decimal.Decimal {
+	if opt.IsNull {
+		panic(fmt.Errorf("unexpected null opt decimal %#v", opt))
+	}
+	return opt.Decimal
+}
+
+func printCosts(deltasBySec map[string]*SecurityDeltas, full bool, csvOutDir string) {
```

Comment:
Most of this function looks like it should be in render.go. Though csvOut wouldn't really belong there, so it probably should be split up.

The naming is also a bit confusing, since it doesn't print to the stdout, it's just writing to csv files... I think.

Reply:
I moved the guts to `render.go`, and made it return a CostsTables object which is the Total & Yearly objects.